contrib/AdaEurope2018/rewrite_self_arg.adb

```diff
@@ -0,0 +1,171 @@
+--  Rename controlling parameters of primitive subprograms to "Self". Write the
+--  result in ".new" suffixed files.
+
+with Ada.Text_IO;
+
+with Langkit_Support.Text;
+with Libadalang.Analysis;
+with Libadalang.Rewriting;
+
+with Helpers;
+
+procedure Rewrite_Self_Arg is
+
+   package TIO renames Ada.Text_IO;
+   package LAL renames Libadalang.Analysis;
+   use type LAL.Ada_Node_Kind_Type;
+   package LALR renames Libadalang.Rewriting;
```

Comment:
Plz choose a renaming name that is not the name of a parsing algorithm . IOW TIA